src/tags/tests/repository.rs

```diff
@@ -509,6 +509,99 @@ mod add_implicit_tag_to_files_tests {
     }
 }
 
+mod add_implicit_tags_to_files_tests {
+    use crate::repository::open_connection;
+    use crate::tags::repository::{add_implicit_tags_to_files, get_all_tags_for_file};
+    use crate::test::*;
+
+    #[test]
+    fn adds_multiple_tags_to_multiple_files() {
+        init_db_folder();
+        create_folder_db_entry("parent", None); // id 1
+        create_file_db_entry("file1.txt", Some(1)); // id 1
+        create_file_db_entry("file2.txt", Some(1)); // id 2
+        let tag_id1 = create_tag_db_entry("test_tag1");
+        let tag_id2 = create_tag_db_entry("test_tag2");
+        let con = open_connection();
+        add_implicit_tags_to_files(&[1, 2], &[tag_id1, tag_id2], 1, &con).unwrap();
+        let tags1 = get_all_tags_for_file(1, &con).unwrap();
+        let tags2 = get_all_tags_for_file(2, &con).unwrap();
+        assert_eq!(tags1.len(), 2);
+        assert_eq!(tags2.len(), 2);
+        con.close().unwrap();
+        cleanup();
+    }
+
+    #[test]
+    fn works_with_empty_file_slice() {
+        init_db_folder();
+        create_folder_db_entry("parent", None); // id 1
+        let tag_id = create_tag_db_entry("test_tag");
+        let con = open_connection();
+        add_implicit_tags_to_files(&[], &[tag_id], 1, &con).unwrap();
+        con.close().unwrap();
+        cleanup();
+    }
+
+    #[test]
+    fn works_with_empty_tag_slice() {
```

Comment:
Rename this to detail that no database interactions should happen if tags is empty